pkg/multiboot/ranges.go

```diff
@@ -0,0 +1,261 @@
+// Copyright 2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package multiboot
+
+import (
+	"bytes"
+	"debug/elf"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/u-root/u-root/pkg/kexec"
+)
+
+var ErrNotEnoughSpace = errors.New("Not enough space")
+
+var sizeofMmap = uint32(binary.Size(Mmap{}))
+
+type Segments []kexec.Segment
+
+type rangeType string
+
+const (
+	rangeRAM      rangeType = "System RAM"
+	rangeNVACPI             = "ACPI Non-volatile Storage"
+	rangeACPI               = "ACPI Tables"
+	rangeReserved           = "Reserved"
+)
+
+var rangeTypes = map[rangeType]uint32{
+	rangeRAM:      1,
+	rangeNVACPI:   3,
+	rangeACPI:     3,
+	rangeReserved: 4,
+}
+
+type memoryRange struct {
+	start, end uintptr
+	typ        rangeType
+}
+
+type memoryRanges []memoryRange
+
+type Mmap struct {
+	Size uint32
+
+	BaseAddr uint64
+	Length   uint64
```

Comment:
A brief description explaining the difference between `Size` and `Length` would be helpful.

Reply:
added comments